user.py

```diff
@@ -91,19 +91,28 @@ def confirm_token(token, expiration=3600):  # skip in tests for now
     return email
 
 
+def check_password_hash(pw_hash, salt, password):
+    password_hash = hashlib.pbkdf2_hmac(HASH_FUNCTION, password, salt.encode('utf-8'), ITERATIONS).hex()
+    if pw_hash == password_hash:
+        return True
+    return False
+
+
 @users_bp.route('/register/', methods=['GET', 'POST'])
 def register():
     """Registration of a user"""
     form = RegistrationForm(request.form)
     if request.method == 'POST':
         if form.validate_on_submit():
             email = form.email.data
-            password = bcrypt.generate_password_hash(form.password.data.encode())
+            salt = generate_salt()
+            password = hashlib.pbkdf2_hmac(HASH_FUNCTION, form.password.data.encode('utf-8'),
+                                           salt.encode('utf-8'), ITERATIONS).hex()
             user_details = dict(
                 confirmed=False
             )
             with db.connect(DEV_DB_NAME) as db_instance:
-                user = UserModel(db_instance, email, password, **user_details)
+                user = UserModel(db_instance, email, password, salt, **user_details)
```

Comment:
hash(password) can not be stored in the system.

Only hash(hash(password)) can be stored.

This is because hash(password) is used as a key to encrypt the symmetric key we use to encrypt files.